pandas/_libs/reduction.pyx

```diff
@@ -370,7 +370,17 @@ def apply_frame_axis0(object frame, object f, object names,
                     mutated = True
             except AttributeError:
                 # `piece` might not have an index, could be e.g. an int
-                pass
+                # By definition, we are not a transform, so set mutated
```

Comment:
@WillAyd I wanted to highlight these two changes:

1. Returning a non-indexed object (e.g. scalar) is defined to be mutated, I think a bugfix
2. I've expanded the definition of mutated to apply to both axes for dataframes (here and https://github.com/pandas-dev/pandas/pull/34998/files#diff-8c0985a9fca770c2028bed688dfc043fR905)

Reply:
Is there a brief explanation of why this is required? I don't have a strong handle on what we define as "mutated" here and in groupby generally

Reply:
I'm also uncertain here, but I *think* that on master `mutated` means "has the same index. On master, we use `mutated` to inform how the result should be wrapped (whether to include the group keys in the index). See https://github.com/pandas-dev/pandas/pull/34998/files#diff-6d2a4be351f3d7ec35b9c6696a615242L892.

The goal here is to rely on the user-provided `group_keys` instead.